runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala

```diff
@@ -45,7 +45,7 @@ object GrpcClientSettings {
     val defaultServiceConfig = actorSystem.settings.config.getConfig("akka.grpc.client").getConfig("\"*\"")
       .withValue("service-name", ConfigValueFactory.fromAnyRef(serviceName))
       .withValue("port", ConfigValueFactory.fromAnyRef(defaultPort))
-      .withValue("service-discovery-mechanism", ConfigValueFactory.fromAnyRef(serviceDiscoveryMechanism))
+      .withValue("service-discovery.mechanism", ConfigValueFactory.fromAnyRef(serviceDiscoveryMechanism))
```

Comment:
How would `discoverService` be used? Would it make more sense to replace it by something like `connectTo(name: String, discovery: SimpleServiceDiscovery, portName: Option[String], protocol: Option[String])` (and introduce `withServiceDiscovery`, `withServicePortName` and `withServiceProtocol` to implement it safely)?

Reply:
That is a good idea. I'm fighting that exactly while working on https://github.com/lagom/akka-grpc-lagom-quickstart-scala

Reply:
`connectTo(name: String, discovery: SimpleServiceDiscovery, portName: Option[String], protocol: Option[String])` looks better than passing a service discovery name around.